sdk_v2/kubeflow/training/types/types.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2024 The Kubeflow Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from dataclasses import dataclass, field
+from typing import Callable, Dict, List, Optional
+
+from kubeflow.training.constants import constants
+
+
+# Representation for the Training Runtime.
+@dataclass
+class Runtime:
+    name: str
+    phase: str
+    device: str
+    device_count: str
+
+
+# Representation for the TrainJob component.
+@dataclass
+class Component:
+    name: str
+    status: str
+    device: str
+    device_count: str
+    pod_name: str
+
+
+# Representation for the TrainJob.
+# TODO (andreyvelich): Discuss what fields users want to get.
+@dataclass
+class TrainJob:
+    name: str
+    runtime_ref: str
+    creation_timestamp: str
+    components: List[Component]
+    status: Optional[str] = "Unknown"
+
+
+# Configuration for the Lora to configure parameter efficient fine-tuning.
+@dataclass
+class LoraConfig:
+    r: Optional[int] = field(
+        default=None, metadata={"help": "Lora attention dimension"}
+    )
+    lora_alpha: Optional[int] = field(default=None, metadata={"help": "Lora alpha"})
+    lora_dropout: Optional[int] = field(default=None, metadata={"help": "Lora dropout"})
+
+
+@dataclass
+class FineTuningConfig:
+    # TODO (andreyvelich): Add more configs once we support them, e.g. QLoRA.
+    peft_config: Optional[LoraConfig] = None
+
```

Comment:
Do we have plan to support some framework-specific policies such as FSDP in PyTorch and ZeRO in DeepSeed? If so, I think it would be great to add a new field named `sharding_config` to specify these features:)

Or we could implement them when designing trainer for LLMs in #2321 . WDYT? @andreyvelich

Reply:
Yeah, I think we should.
We might want to implement them once we have initial support for LLM Trainer in upstream.
We are still discussing how the LLM Trainer should look like.